### PR TITLE
[hkg] Add additional haskell libraries to support

### PR DESCRIPTION
**Pull request by crogers1**
toolstack functionality.

Signed-off-by: Chris Rogers <crogers122@gmail.com>

### CODE REVIEW

**Comment by crogers1** [comment]
@jandryuk going to let this one sit here for a bit, the code that would use the libraries is going to take longer to decouple and open PRs for.

**Comment by tsirakisn** [comment]
@crogers1 Any reason we're waiting to merge this one? I don't see an issue with adding recipe files, we aren't adding any of these packages to the images. The other repos are gonna take a long time to disect.

**Comment by crogers1** [comment]
> @crogers1 Any reason we're waiting to merge this one? I don't see an issue with adding recipe files, we aren't adding any of these packages to the images. The other repos are gonna take a long time to disect.

I suppose you have a point. @jandryuk if you're willing to have the recipes merged in but unused until later, it would reduce our open PR count.

**Comment by jandryuk** [comment]
@crogers1 suggested the usual approach of waiting for to merge support pieces until they are actually used.  That keeps things that may never be used from getting in the tree.

@tsirakisn I also realize that cross repo dependencies complicate development and merging this may make it easier.

This commit adds 4 new recipes.  How about you expand the commit message to state which packages and which features they will support?  That'll document why they were added and would help track if they can be removed for whatever reason in the future.

**Comment by crogers1** [comment]
I can do that.

**Comment by crogers1** [comment]
Updated commit with descriptions of what the libraries are used for.  In my review, it is possible to re-factor the code to work with other functions and thus avoid including these libraries, but I believe the complexity of the code would increase where these functions are being used.

**Comment by jandryuk** [comment]
Sorry, I wasn't clear.  What other OpenXT pieces are going to use these libraries and why?  Guessing from what you wrote, you have a XenMgr change to do base64 with certificates?

**Comment by crogers1** [comment]
> Sorry, I wasn't clear. What other OpenXT pieces are going to use these libraries and why? Guessing from what you wrote, you have a XenMgr change to do base64 with certificates?

I would be able to open source XenMgr functions that can decode a base64 x509 cert and parse it for various information such as the expiry date, because that can be considered general functionality.  The remainder of what I use that code for I would not be able to open source at this time.  The benefit to OpenXT and the community would simply be expanding what the toolstack XenMgr API is capable of.  

The other libraries provided in this PR follow a similar reasoning.  If that's too thin to go on, I could refactor the code to _not_ use these libraries (would require some dev work though, some of the functions we get here are pretty handy), and then the various XenMgr commits could be PR'd without needing these new libraries.

Third option, which I'm personally starting to lean toward, is to close this PR, and as the XenMgr changes get PRd, add in these libraries at that time to tie them together and more easily talk about what functionality they provide, pointing to specific examples in the code.  Niko's and my process so far has been to sync changes in repositories with a small diff first before moving onto the larger diff repositories, and this repo was near the top of that list.

**Comment by jandryuk** [comment]
They are just standalone libraries, so I guess it's fine to just merge this so they are available when needed.

**Review by jandryuk — APPROVED** [review]
Will merge tomorrow.

File: recipes-devtools/hackage/hkg-base64-bytestring_1.0.0.1.bb
Comment (jandryuk) [inline recipes-devtools/hackage/hkg-base64-bytestring_1.0.0.1.bb]: There is already an `inherit hackage` above.

Comment (crogers1) [reply recipes-devtools/hackage/hkg-base64-bytestring_1.0.0.1.bb]: Fixed.  Slipped through.  I need to visit the eye doc :joy:

**Pull request merged**
